### PR TITLE
3278/civ 4203

### PR DESCRIPTION
**Pull request by dtortolaV1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-4203

### Change description ###
Built as a branch from the ones for https://tools.hmcts.net/jira/browse/CIV-3278
Changes are to use the new dropdown for spec journey preferred court location instead of the free text field

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: charts/civil-ccd/values.preview.template.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ postgresql:
 
 java:
   applicationPort: 4000
-  image: 'hmctspublic.azurecr.io/civil/service:latest'
+  image: 'hmctspublic.azurecr.io/civil/service:pr-1334'
```
Comment (sherlynkhaw) [inline charts/civil-ccd/values.preview.template.yaml:27]: remove this change please

File: ccd-definition/AuthorisationCaseField/AuthorisationCaseField.json [hunk 2]
```diff
@@ -1803,7 +1805,10 @@
     "AccessControl": [
       {
         "UserRoles": [
-          "[RESPONDENTSOLICITORTWO]"
+          "[RESPONDENTSOLICITORONE]",
+          "[RESPONDENTSOLICITORTWO]",
```
Comment (sherlynkhaw) [inline ccd-definition/AuthorisationCaseField/AuthorisationCaseField.json:1809]: "[RESPONDENTSOLICITORONE]","[RESPONDENTSOLICITORONESPEC]" should be removed


I'm not sure about "[RESPONDENTSOLICITORONESPEC]" though?

File: ccd-definition/CaseEvent/User/UserEventsLRspec.json [hunk 3]
```diff
@@ -64,9 +64,9 @@
     "PreConditionState(s)": "AWAITING_RESPONDENT_ACKNOWLEDGEMENT",
     "PostConditionState": "*",
     "SecurityClassification": "Public",
-    "CallBackURLAboutToStartEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/cases/callbacks/version/V_1/about-to-start",
+    "CallBackURLAboutToStartEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/cases/callbacks/version/V_2/about-to-start",
```
Comment (sherlynkhaw) [inline ccd-definition/CaseEvent/User/UserEventsLRspec.json:67]: please create a prod file and a non prod file for this
1. move DEFENDANT_RESPONSE_SPEC and CLAIMANT_RESPONSE_SPEC old (master) code into prod file
2. move DEFENDANT_RESPONSE_SPEC and CLAIMANT_RESPONSE_SPEC new code into nonprod file
3. remove DEFENDANT_RESPONSE_SPEC and CLAIMANT_RESPONSE_SPEC from base file

File: ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json [hunk 4]
```diff
@@ -264,18 +264,7 @@
...
-    "ListElementCode": "applicantRequestedCourt",
-    "EventElementLabel": " ",
-    "HintText": " ",
-    "FieldDisplayOrder": 1,
-    "DisplayContext": "MANDATORY",
-    "FieldShowCondition": "requestHearingAtSpecificCourt = \"DO_NOT_SHOW_IN_UI\""
-  },
-  {
-    "ID": "RequestedCourt",
+    "ID": "RespondToClaim",
```
Comment (sherlynkhaw) [inline ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json:267]: Should this stay as 'RequestedCourt'?

File: ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json [hunk 5]
```diff
@@ -285,25 +274,15 @@
...
-    "CaseEventID": "CLAIMANT_RESPONSE_SPEC",
-    "CaseFieldID": "applicant1DQRequestedCourt",
-    "ListElementCode": "courtFinderLabel",
-    "EventElementLabel": "<a target=\"_blank\" href=\"https://courttribunalfinder.service.gov.uk/search/address\">Court and tribunals finder (opens in new tab)</a> Use the 3 digit court location code, not the court name",
-    "FieldDisplayOrder": 2,
-    "DisplayContext": "READONLY"
-  },
-  {
-    "ID": "RequestedCourt",
+    "ID": "RespondToClaim",
```
Comment (sherlynkhaw) [inline ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json:277]: Should this stay as 'RequestedCourt'?

File: ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json [hunk 6]
```diff
@@ -285,25 +274,15 @@
...
     "CaseFieldID": "applicant1DQRequestedCourt",
-    "ListElementCode": "responseCourtCode",
-    "EventElementLabel": "Court location code",
+    "ListElementCode": "responseCourtLocations",
     "FieldDisplayOrder": 3,
     "DisplayContext": "MANDATORY"
   },
   {
-    "ID": "RequestedCourt",
+    "ID": "RespondToClaim",
```
Comment (sherlynkhaw) [inline ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json:285]: Should this stay as 'RequestedCourt'?

File: ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json (hunk 4 above)
Comment (sherlynkhaw) [reply ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json:267]: these would also have to go into prod/non prod files

File: ccd-definition/AuthorisationCaseField/AuthorisationCaseField.json (hunk 2 above)
Comment (dtortolaV1) [reply ccd-definition/AuthorisationCaseField/AuthorisationCaseField.json:1809]: 1v2 spec same solicitor still needs both 1 and 2 fields, that's why respondentsolicitoronespec is needed.
I remove the unspec ones

File: ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json [hunk 7]
```diff
@@ -284,24 +273,6 @@
...
-    "ID": "RequestedCourt",
-    "CaseEventID": "CLAIMANT_RESPONSE_SPEC",
-    "CaseFieldID": "applicant1DQRequestedCourt",
-    "ListElementCode": "courtFinderLabel",
-    "EventElementLabel": "<a target=\"_blank\" href=\"https://courttribunalfinder.service.gov.uk/search/address\">Court and tribunals finder (opens in new tab)</a> Use the 3 digit court location code, not the court name",
-    "FieldDisplayOrder": 2,
-    "DisplayContext": "READONLY"
-  },
-  {
-    "ID": "RequestedCourt",
```
Comment (sherlynkhaw) [inline ccd-definition/CaseEventToComplexTypes/ClaimantResponseLRspec.json:297]: might need to keep this one

File: ccd-definition/CaseEvent/User/UserEventsLRspec.json [hunk 8]
```diff
@@ -64,8 +64,8 @@
     "PreConditionState(s)": "AWAITING_RESPONDENT_ACKNOWLEDGEMENT",
     "PostConditionState": "*",
     "SecurityClassification": "Public",
-    "CallBackURLAboutToStartEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/cases/callbacks/about-to-start",
-    "CallBackURLAboutToSubmitEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/cases/callbacks/about-to-submit",
+    "CallBackURLAboutToStartEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/cases/callbacks/version/V_1/about-to-start",
```
Comment (sherlynkhaw) [inline ccd-definition/CaseEvent/User/UserEventsLRspec.json:67]: should these have their own -prod/-nonprod files too?

File: ccd-definition/ComplexTypes/DQ/RequestedCourt.json [hunk 9]
```diff
@@ -21,5 +21,13 @@
     "ElementLabel": "Briefly explain your reasons",
     "FieldShowCondition": "requestHearingAtSpecificCourt = \"Yes\"",
     "SecurityClassification": "Public"
+  },
+  {
```
Comment (sherlynkhaw) [inline ccd-definition/ComplexTypes/DQ/RequestedCourt.json:25]: why is this in the base file? i think it should be kept in the nonprod file?

File: ccd-definition/ComplexTypes/DQ/RequestedCourt-nonprod.json [hunk 10]
```diff
@@ -7,14 +7,6 @@
     "FieldShowCondition": "requestHearingAtSpecificCourt = \"DO NOT SHOW IN UI\"",
     "SecurityClassification": "Public"
   },
-  {
```
Comment (sherlynkhaw) [inline ccd-definition/ComplexTypes/DQ/RequestedCourt-nonprod.json:10]: don't think this should be removed

**Review by sherlynkhaw — CHANGES_REQUESTED** [review]

**Review by sherlynkhaw — APPROVED** [review]

**Review by sherlynkhaw — APPROVED** [review]

File: ccd-definition/AuthorisationCaseField/AuthorisationCaseField.json [hunk 11]
```diff
@@ -1784,7 +1784,8 @@
     "AccessControl": [
       {
         "UserRoles": [
-          "[RESPONDENTSOLICITORONE]"
+          "[RESPONDENTSOLICITORONE]",
+          "[RESPONDENTSOLICITORONESPEC]"
```
Comment (alberto-hortiguela) [inline ccd-definition/AuthorisationCaseField/AuthorisationCaseField.json:1788]: The AP will get merged today needing some rework here as we will not be using the traditional CaseRoles
https://github.com/hmcts/civil-ccd-definition/pull/1267/files

**Pull request closed without merging**